# Fix ungrib/src/cio.c compilation error with newer C compilers
# due to functions not being declared with "int"
# FYI, cio.c last modified date is 15 years ago, as of this writing.
# By Thang Ha

diff -Nru WPS-4.5.orig/ungrib/src/cio.c WPS-4.5/ungrib/src/cio.c
--- WPS-4.5.orig/ungrib/src/cio.c	2024-06-08 21:45:08.510489331 -0500
+++ WPS-4.5/ungrib/src/cio.c	2024-06-08 21:39:22.235745617 -0500
@@ -49,7 +49,7 @@
 
 /* ****************************************************************** */
 
-c_open(unit, nunit, name, mode, err, oflag)
+int c_open(unit, nunit, name, mode, err, oflag)
  /*
   * unit  = Fortran unit number 
   * nunit = UNIX file descriptor associated with 'unit'
@@ -114,7 +114,7 @@
 }
 
 /* ****************************************************************** */
-bn_seek(fd, bread, mode, iprint)
+int bn_seek(fd, bread, mode, iprint)
 
 /*  Move the read/write file pointer
        fd     : Unix file descriptor.
@@ -152,7 +152,7 @@
 
 /* ****************************************************************** */
 
-bn_read(fd, buf, nbuf, bread, ios, idiag)
+int bn_read(fd, buf, nbuf, bread, ios, idiag)
  /*
   * fd = UNIX file descriptor number (NOT a Fortran unit) 
   * buf = area into which to read 
@@ -198,7 +198,7 @@
 
 /* ****************************************************************** */
 
-bnwrit_(fd, buf, nbuf, bwritten, err, idiag)
+int bnwrit_(fd, buf, nbuf, bwritten, err, idiag)
     int            *fd, *nbuf, buf[], *bwritten, *err, *idiag;
 
  /*
@@ -232,7 +232,7 @@
 
 /* ****************************************************************** */
 
-c_close(nunit, iprint, err)
+int c_close(nunit, iprint, err)
 /*
 Close a C (UNIX?) file descriptor:
   nunit  : (INPUT)  : The C (UNIX?) file descriptor to close.
